File: src/math/sum.ts
/*
  // regular
  export function sum(n: number): number {
    let result: number = 0
    for (let i: number = 0; i <= n; i++) {
      result += i
    }
    return result
  }
*/

// recursion
export function sum(n: number): number {
  if (n >= 0) {
    return n += sum(n - 1)
  } else {
    return 0
  }
}

console.log(sum(10))

File: src/math/sum.cpp
#include <iostream>

using namespace std;

//
int sumA(int num) {
    int result = 0;
    for (int i = 1; i <= num; i++) {
        result += i;
    }
    return result;
}

//
int sumB(int num) {
    int result = 0;
    while (num > 0) {
        result += num--;
    }
    return result;
}

// recursion
int sumC(int num) {
    if (num > 0) {
        return num += sumC(num - 1);
    } else {
        return 0;
    }
}

int main() {
    cout << "Function sumA: " << sumA(10) << endl;
    cout << "Function sumB: " << sumB(10) << endl;
    cout << "Function sumC: " << sumC(10) << endl;
    return 0;
}